docs/psf_spec/high_level_overview.rst

```diff
@@ -0,0 +1,244 @@
+Fitter_ when minimizing the fit between the model and the data.
+
+
+.. _noise:
+
+Noise Description
+^^^^^^^^^^^^^^^^^
+The ``noise`` block within the PSF fitting process is, in much the same way as the ``Fitter``
+block, a separate ``callable`` function mapping the relationship between the pixel values in
+the input ``image`` and the corresponding uncertainty -- in units of standard deviations -- of
+the pixel. It must therefore accept a two-dimensional array and return an array of the same
+shape as ``image``. Alternatively it can be overloaded to indicate that the uncertainty array
+was passed along with ``image`` via an `~astropy.nddata.NDData` instance. A more detailed
+description of this block is available :doc:`here <noise_data>`.
+
+
+.. _Background Estimator:
+
+Background Estimator
+^^^^^^^^^^^^^^^^^^^^
+The ``image`` on which source extraction and evaluation is to be performed is not necessarily
+assumed to have had any pre-processing applied to it, and thus it may be necessary to account
+for the counts of the image background when handling PSF fitting. This block handles
+the determination of the background levels, avoiding high count peaks to calculate the typical
+count of an otherwise empty pixel. The block must therefore accepts the two-dimensional ``image``
+and returns either a single value -- the background count of the entire image, as calculated
+by the criteria of the specific block implementation -- or a 2D array of values, allowing for
+the possibility of varying background counts across the image. More information on this block
+can be found :doc:`here <background_estimator>`.
+
+
+.. _Culler and Ender:
+
+Culler and Ender
+^^^^^^^^^^^^^^^^
+Only implemented in iterative fitting classes, part of the extension to
+`~photutils.psf.BasicPSFPhotometry`, this block is the final step of a single iteration. While
+a maximum number of iterations can be specified for, e.g., 
+`~photutils.psf.IterativelySubtractedPSFPhotometry`, this block -- specifically, the ender half
+-- assesses whether all sources have been found within the ``image``. If no new sources have
+been found within an iteration, fitting can be stopped prematurely, avoiding wasteful computational
+time. Similarly, the culler aspect of this block examines sources found within an iteration for
+quality; it should calculate some goodness-of-fit criterion and reject sources picked up by the
+``Finder`` but fall below a given threshold. This block therefore accepts the ``stars``
+output `~astropy.table.Table` and returns a new `~astropy.table.Table` with low quality sources
+-- such as cosmic ray hits present in the image -- removed as well as a boolean flag indicating
+whether the iterative fitting process has reached a conclusion and can be terminated. Please refer
+to the :doc:`documentation <culler_and_ender>` for further details.
+
+
+.. _Single Object Model:
+
+Single Object Model
```

Comment:
This (along with the scene maker) are *optional*, right?  That is, if not present, it means just use the PSF model directly?  In that case you should say that in both descriptions.

Reply:
I've added language to this effect in the text, but eventually I don't think they'll be entirely optional, in that I think the block will always have to be used, but it can have convenient "do nothing" defaults. I think it's tied to the ``object_type`` flag that will eventually have to be thrown around in the ``astropy.table.Table`` objects: if that (by default) is something like ``point source`` then ``Single Object Model`` simply returns the input PSF model (i.e., f * &#948; = f) and ``Scene Maker`` just returns the input table of objects (i.e., no input sources are an extended source split apart).

This *effectively* means they're optional, as the user can just ignore that the more complex functionality exists and default parameters unchanged give the same answer as if they weren't there, but I think that's easier than having ``if use_som: new_model = single_object_model(psf)`` or equivalent.